backend/src/services/auto_trader.py

```
@@ -33,238 +31,203 @@ def __init__(self, socketio=None, data_fetcher_service=None, ai_analysis_service
+            logger.error(f"Error in scan and buy: {e}")
 
     async def _monitor_and_sell(self):
-        logger.info("AutoTrader: Monitoring owned tokens for sell opportunities...")
         tokens_to_remove = []
         for token_address, details in self.owned_tokens.items():
             try:
                 current_token_data = await self.data_fetcher_service.get_token_by_address(token_address)
-                if not current_token_data:
-                    logger.warning(f"AutoTrader: Owned token {details.get('symbol', token_address)} not found in data fetcher. Skipping sell monitoring.")
-                    continue
+                if not current_token_data: continue
```

Ruff (0.15.5)

[error] Multiple statements on one line (colon)

(E701)